store/store.go

```diff
@@ -2089,29 +2092,42 @@ func (s *Store) SnapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 	}
 }
 
+func genInstanceKey(curSnap *CurrentSnap, seed string) string {
+	_, snapInstanceKey := snap.SplitInstanceName(curSnap.InstanceName)
+
+	if snapInstanceKey == "" {
+		return curSnap.SnapID
+	}
+
+	// due to privacy concerns, avoid sending the local names to the
+	// backend, instead hash the snap ID and instance key together
+	h := crypto.SHA256.New()
+	// TODO parallel-install: include seed
+	h.Write([]byte(curSnap.SnapID))
+	h.Write([]byte(snapInstanceKey))
+	h.Write([]byte(seed))
+	enc := base64.RawURLEncoding.EncodeToString(h.Sum(nil))
+	return fmt.Sprintf("%s-%s", curSnap.SnapID, enc)
+}
+
 func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, actions []*SnapAction, user *auth.UserState, opts *RefreshOptions) ([]*snap.Info, error) {
 
 	// TODO: the store already requires instance-key but doesn't
 	// yet support repeating in context or sending actions for the
 	// same snap-id, for now we keep instance-key handling internal
 
+	requestSeed := ""
```

Comment:
I wonder if we should error in case  len(currentSnaps) != 0 but there is no seed/salt set